src/components/calcite-picker-row/calcite-picker-row.tsx

```diff
@@ -0,0 +1,152 @@
+import { Component, Element, Event, EventEmitter, Host, Method, Prop, h } from "@stencil/core";
+import {
+  checkSquare16,
+  circle16,
+  circleFilled16,
+  handleVertical24,
+  square16,
+  trash16
+} from "@esri/calcite-ui-icons";
+import { CSS } from "./resources";
+import CalciteIcon from "../utils/CalciteIcon";
+
+@Component({
+  tag: "calcite-picker-row",
+  styleUrl: "./calcite-picker-row.scss",
+  shadow: true
+})
+export class CalcitePickerRow {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Prop({ reflect: true }) editing = false;
+
+  @Prop({ reflect: true }) icon: "square" | "circle" | "grip" | null = null;
+
+  @Prop() metadata: object;
+
+  @Prop({ reflect: true, mutable: true }) selected = false;
+
+  @Prop({ reflect: true }) textHeading: string;
+
+  @Prop({ reflect: true }) textDescription: string;
+
+  @Prop({ reflect: true }) value: string;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() calcitePickerRowToggled: EventEmitter;
+  @Event() calcitePickerRowDeleted: EventEmitter;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method() async toggle(shiftPressed: boolean) {
+    this.selected = !this.selected;
+    this.calcitePickerRowToggled.emit({
+      row: this.el,
+      value: this.value,
+      selected: this.selected,
+      shiftPressed
+    });
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  iconClickHandler(e) {
+    this.toggle(e.shiftKey);
+  }
+
+  secondaryActionContainerClickHandler(e) {
+    e.stopPropagation();
+  }
+
+  deleteRow() {
+    this.calcitePickerRowDeleted.emit({ row: this.el, value: this.value });
```

Comment:
I thought we were removing the delete stuff?

Reply:
separate PR.